pandas/core/ops.py

```diff
@@ -6,31 +6,30 @@
 # necessary to enforce truediv in Python 2.X
 from __future__ import division
 import operator
-import warnings
+
 import numpy as np
 import pandas as pd
-import datetime
 
 from pandas._libs import (lib, index as libindex,
-                          tslib as libts, algos as libalgos, iNaT)
+                          algos as libalgos)
 
 from pandas import compat
 from pandas.util._decorators import Appender
 
 from pandas.compat import bind_method
 import pandas.core.missing as missing
 
-from pandas.errors import PerformanceWarning, NullFrequencyError
```

Comment:
lgtm. are there any of these PerformanceWarnings uncaught?

Reply:
Pretty sure we got them all while implementing the Index versions of these ops.